### PR TITLE
feat: Combine webhooks and controllers into a single binary

### PR DESCRIPTION
**Pull request by ellistarn**
*Issue #, if available:*

*Description of changes:*

Jonathan and I discovered that we could avoid the --kubeconfig conflict by relying on controllerruntime libraries to initialize the `rest.Config` for both webhooks and the controller. This enables us to collapse these into a single pod, simplifying operations and release infrastructure.

Some caveats:
- Helm chart flags to configure the webhook will no longer work
- Webhook logging has been turned down to ERROR, since INFO/Debug is so spammy.

Partner PR: https://github.com/aws/karpenter/pull/2727

*Testing:*
* `make e2etest`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cloudprovider/types.go [hunk 1]
```diff
@@ -32,10 +32,10 @@ import (
 type Context struct {
 	context.Context
 
-	ClientSet     *kubernetes.Clientset
-	KubeClient    client.Client
-	EventRecorder record.EventRecorder
-	Clock         clock.Clock
+	KubernetesInterface kubernetes.Interface
```
Comment (jonathan-innis) [inline pkg/cloudprovider/types.go:35]: `KubeInterface`?

File: pkg/controllers/controllers.go [hunk 2]
```diff
@@ -35,20 +41,30 @@ func init() {
 	metrics.MustRegister() // Registers cross-controller metrics
 }
 
-func GetControllers(ctx operator.Context, cluster *state.Cluster, settingsStore settingsstore.Store, cloudProvider cloudprovider.CloudProvider) []controller.Controller {
-	provisioner := provisioning.NewProvisioner(ctx, ctx.KubeClient, ctx.Clientset.CoreV1(), ctx.EventRecorder, cloudProvider, cluster, settingsStore)
+func NewControllers(
+	ctx context.Context,
+	clock clock.Clock,
+	kubeClient client.Client,
+	clientset kubernetes.Interface,
```
Comment (jonathan-innis) [inline pkg/controllers/controllers.go:48]: `kubeInterface`?

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+			ctx = injection.WithConfig(ctx, config)
+			ctx = injection.WithOptions(ctx, *opts)
+			return ctx
+		},
+	})
+	manager = lo.Must(manager, err, "failed to setup manager")
+	if opts.EnableProfiling {
+		registerPprof(manager)
+	}
+	lo.Must0(manager.GetFieldIndexer().IndexField(ctx, &v1.Pod{}, "spec.nodeName", func(o client.Object) []string {
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:128]: Didn't know there was `Must0`! That's awesome!

**Review by jonathan-innis — DISMISSED** [review]
Really, really neat +1,000,000 🎉

File: pkg/controllers/controllers.go (hunk 2 above)
Comment (jonathan-innis) [reply pkg/controllers/controllers.go:48]: Maybe some consistent name for `kubernetes.Interface` would be good

Comment (ellistarn) [reply pkg/controllers/controllers.go:48]: Ah -- I meant to make this kubernetesInterface everywhere. It brainlessly concatenate package and type.

**Review by jonathan-innis — APPROVED** [review]
🚀

**Pull request merged**